### PR TITLE
Add Support for BigQuery COUNTIF Function

### PR DESCRIPTION
**Pull request by aryeh-looker**
Adds support for the BigQuery `COUNTIF` function https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#countif

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlJdbcFunctionCall.java [hunk 1]
```diff
@@ -752,6 +752,7 @@ private JdbcToInternalLookupTable() {
               return super.createCall(pos, operands);
             }
           });
+      map.put("COUNTIF", simple(SqlLibraryOperators.COUNTIF));
```
Comment (aryeh-looker) [inline core/src/main/java/org/apache/calcite/sql/SqlJdbcFunctionCall.java:755]: We likely could implement calcite support for this function, as opposed to merely using `simple`. Not sure on convention — i.e. if changes like this are accepted upstream for mere parseability.

Comment (julianhyde) [reply core/src/main/java/org/apache/calcite/sql/SqlJdbcFunctionCall.java:755]: Probably don't need COUNTIF in `SqlJdbcFunctionCall`. This class is just for the `{fn COUNTIF()}` syntax, which few people use.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -477,4 +477,16 @@ private SqlLibraryOperators() {
...
+   * zero input rows, or if expression evaluates to FALSE or NULL for all rows.
+    */
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction COUNTIF =
+      new SqlFunction("COUNTIF", SqlKind.OTHER_FUNCTION,
+          ReturnTypes.BIGINT_NULLABLE,
+          null,
+          OperandTypes.BOOLEAN,
+          SqlFunctionCategory.SYSTEM);
+
```
Comment (julianhyde) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:491]: I'm a bit surprised that you were able to get this to work as a sub-class of SqlFunction. I'd have thought SqlAggFunction.

One useful test would be a validator test something like `select sum(sal), countif(sal > 100) from emp group by deptno`. Will pass if `countif` is correctly being registered as an agg function. Will fail if `countif` is a regular function.

**Review by julianhyde — COMMENTED** [review]
I don't think this will pass muster as an aggregate function. Additional tests a required to prove it works. Since this is the first time anyone has added an aggregate function to `SqlLibraryOperators` it probably won't work first time; I am happy to pair-program it with you or take over the hard parts.

Also, I just logged https://issues.apache.org/jira/browse/CALCITE-4335. Can you prefix the commit message with "[CALCITE-4335] ".

**Comment by aryeh-looker** [comment]
Replaced due to GitHub losing track of what's going on during rebase https://github.com/apache/calcite/pull/2235

**Pull request closed without merging**